vortex-datafusion/src/persistent/opener.rs

```diff
@@ -206,28 +201,38 @@ impl FileOpener for VortexOpener {
                 );
             }
 
-            let filter = filter
+            let predicate = predicate
                 .and_then(|f| {
                     let exprs = split_conjunction(&f)
                         .into_iter()
+                        .filter_map(|e| {
+                            if is_dynamic_physical_expr(&e) {
+                                e.snapshot().ok().flatten()
+                            } else {
+                                Some(Arc::clone(e))
+                            }
```

Comment:
Not sure if it's possible or worth it but you could do this more lazily, or wrap the DataFusion dynamic filter in a Vortex dynamic filter and call `DynamicFilterPhysicalExpr::current()` or `PhysicalExpr::snapshot()` on each RecordBatch (that would be more overhead but also will kick in sooner on scans that touch very large files).

Reply:
I roughly tried to do that in some old PR, but I don't remember why I stopped pushing it. This seems like a good starting point, and we can experiment with it in the future to see which one is better.